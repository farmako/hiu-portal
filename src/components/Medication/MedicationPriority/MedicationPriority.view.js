import React from 'react';
import * as PropTypes from 'prop-types';

const MedicationPriority = (props) => {
  const { mr } = props;
  if (mr.priority) {
    return (
      <li>
        <span>
          Priority:
          {mr.priority}
        </span>
      </li>
    );
  }
  return null;
};

MedicationPriority.propTypes = {
  mr: PropTypes.shape({ priority: PropTypes.string }),
};

MedicationPriority.defaultProps = {
  mr: {
    priority: '',
  },
};

export default MedicationPriority;
