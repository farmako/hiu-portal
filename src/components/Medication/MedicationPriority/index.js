import MedicationPriority from './MedicationPriority.view';

export default MedicationPriority;
