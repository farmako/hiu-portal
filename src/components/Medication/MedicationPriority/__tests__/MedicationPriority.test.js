import React from 'react';
import {shallow} from "enzyme";
import MedicationPriority from "../MedicationPriority.view";

describe('MedicationPriority', () => {
    const mockData = {
        priority:'Medication priority'
    };
    const wrapper = shallow(
      <MedicationPriority mr={mockData} />
    );

    it('should render properly', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("li").exists()).toEqual(true);
    });

    it('should not render anything if data is undefinde', () => {
        wrapper.setProps({mr:undefined});
        expect(wrapper.find("li").exists()).toEqual(false);
    });
});