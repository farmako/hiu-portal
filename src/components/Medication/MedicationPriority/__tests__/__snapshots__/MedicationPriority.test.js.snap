// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MedicationPriority should render properly 1`] = `
<li>
  <span>
    Priority:
    Medication priority
  </span>
</li>
`;
